README.md

```diff
@@ -50,6 +50,26 @@
 
 Readers already familiar with effect systems may wish to start with the [usage](#usage) instead.
 
+<!--
+Setup, hidden from the rendered markdown.
+
+```haskell
+{-# LANGUAGE ConstraintKinds, FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, TypeApplications, UndecidableInstances #-}
+module Main (module Main) where
+
+import Control.Algebra
+import Control.Carrier.Lift
+import Control.Carrier.Reader
+import Control.Carrier.State.Strict
+import Control.Effect.Writer
+import Control.Monad.IO.Class (liftIO)
+import qualified Control.Monad.State.Class as MTL
+
+main :: IO ()
+main = pure ()
+```
+-->
```

Comment:
Step two: we do any setup that was left implicit before; for this document, I’ve opted to embed it in an HTML comment so that it’s still processed but not visible in the rendered markdown.